### PR TITLE
Fhir Client fixes

### PR DESCRIPTION
**Pull request by sjpadgett**
- patient service getOne always returns a 404.
@yashrajbothra you might want to note this commit.

### CODE REVIEW

**Comment by sjpadgett** [comment]
@yashrajbothra and @silvistershebin 
Note in this PR i'm removing all 'patient-' prefix to openemr id's. Hope not to much trouble.

**Comment by yashrajbothra** [comment]
Hey @sjpadgett should we change the names of function to what fhir http://hl7.org/fhir/r4/http.html suggest. Right now its follows our standard names.

**Comment by sjpadgett** [comment]
> Hey @sjpadgett Sir, should we change the names of function to what fhir http://hl7.org/fhir/r4/http.html suggest. Right now its follows our standard names.

give me an example

**Comment by yashrajbothra** [comment]
> > Hey @sjpadgett Sir, should we change the names of function to what fhir http://hl7.org/fhir/r4/http.html suggest. Right now its follows our standard names.
> 
> give me an example

`put()` can replaced with `update()`
`post()` can be replaced with 'create()' etc.

I know names doesn't matter much but it makes following the statndard easy and readable code.

**Comment by sjpadgett** [comment]
> `put()` can replaced with `update()`

I don't know. I'd have to think about that but i'm to tired right now. Maybe others have an opinion but mine is no.

Soon as travis completes i'm bringing so everybody working with refactored code. I think I got all the prefixes but if you see one then remove.

**Comment by yashrajbothra** [comment]
> I don't know. I'd have to think about that but i'm to tired right now. Maybe others have an opinion but mine is no.

No, problem it was just a thought.

> Soon as travis completes i'm bringing so everybody working with refactored code. I think I got all the prefixes but if you see one then remove.

Definitely 👍

File: src/RestControllers/FhirPatientRestController.php [hunk 1]
```diff
@@ -32,41 +32,42 @@ public function __construct($pid)
     public function post($fhirJson)
     {
         $data = $this->fhirService->parsePatientResource($fhirJson);
-        
+
         $validationResult = $this->patientService->validate($data);
         $validationHandlerResult = RestControllerHelper::validationHandler($validationResult);
         if (is_array($validationHandlerResult)) {
             return $validationHandlerResult; }
```
Comment (yashrajbothra) [inline src/RestControllers/FhirPatientRestController.php]:
```suggestion
            return $validationHandlerResult;
        }
```

Travis sir will be happy after this 😃

File: src/RestControllers/FhirPatientRestController.php [hunk 2]
```diff
@@ -32,41 +32,42 @@ public function __construct($pid)
...
 
     public function put($pid, $fhirJson)
     {
         $data = $this->fhirService->parsePatientResource($fhirJson);
-        
+
         $validationResult = $this->patientService->validate($data);
         $validationHandlerResult = RestControllerHelper::validationHandler($validationResult);
         if (is_array($validationHandlerResult)) {
             return $validationHandlerResult; }
```
Comment (yashrajbothra) [inline src/RestControllers/FhirPatientRestController.php]:
```suggestion
            return $validationHandlerResult;
        }
```

File: src/Services/FhirResourcesService.php [hunk 3]
```diff
@@ -97,13 +99,18 @@ public function createPatientResource($resourceId = '', $data = '', $encode = tr
         }
     }
 
-    public function createPractitionerResource($id = '', $data = '', $encode = true)
+    public function createPractitionerResource($provider_id = '', $encode = true)
     {
+        if(!$provider_id) {
```
Comment (yashrajbothra) [inline src/Services/FhirResourcesService.php]:
```suggestion
        if (!$provider_id) {
```
For Happy Travis 😀

Comment (sjpadgett) [reply src/Services/FhirResourcesService.php]: yeah thanks. forgot to scan last night. too tired and too old:) thanks though

**Comment by sjpadgett** [comment]
@bradymiller My dear friend. My compadre, The finest doctor in the U.S. Can we please stop trying to save the world with Travis? The darn thing takes an hour or more sometimes to run and unfortunately my attention span is only 50 minutes these days!

**Comment by sjpadgett** [comment]
Going to hold on the test client till I add another example.

**Comment by bradymiller** [comment]
Hi Jerry,
You now have a license to ignore travis (ie. ok to bring in PRs before
travis testing completes). The problem we have is a good thing; travis only
lets us run 5 tests at a time (and there are 18 total tests in a run); it
works fine when only have a PR or 2 going. But when have multiple PRs and
commits into the codebase then things get very behind. So, ignore travis (I
can easily fix up PSR2 issues intermittently).

**Comment by yashrajbothra** [comment]
I use VS Code as Editor and there is an extension known as "phpcs". So whenever i do any psr2 mistakes it show me error instantly (now need to run test). Its highly recommended if are travis hater like me 😸.

I can add it in Guide if it hepls

**Comment by sjpadgett** [comment]
> can add it in Guide if it hepls

Be nice if add to guide but I use PHPStorm with all that. It is very rare I have travis issues.

**Comment by sjpadgett** [comment]
oh by the way @robertdown , my license expires next month. Can you help out or do I need to wait until they expire before renewal?

**Pull request merged**